File: BankatiProject_Backend/services/VirtualCardService/src/main/java/com/example/virtualcard/virtualcard/entity/VirtualCard.java
package com.example.virtualcard.virtualcard.entity;


import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@Document
public class VirtualCard {
    @Id
    private String id;

    private String cardNumber;

    private String userId;

    private LocalDateTime expirationDate;

    private String status; // Status : "ACTIVE", "INACTIVE", "EXPIRED"

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private double montant;


}

File: BankatiProject_Front-end/src/app/models/VirtualCard.ts
export interface VirtualCard {
  id: string;
  cardNumber: string;
  userId: string;
  expirationDate: string;
  status: 'ACTIVE' | 'INACTIVE' | 'EXPIRED';
  createdAt: string;
  updatedAt: string;
  montant: string;
}
